devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java

```diff
@@ -335,7 +335,7 @@ private void addSelfWithLocalDeps(Project project, GradleDevModeLauncher.Builder
         if (!visited.add(project.getPath())) {
             return;
         }
-        final Configuration compileCp = project.getConfigurations().findByName(JavaPlugin.COMPILE_CLASSPATH_CONFIGURATION_NAME);
+        final Configuration compileCp = project.getConfigurations().findByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME);
```

Comment:
There could also be a compile-only config which is relevant in dev mode, e.g. lombok. Could you check please?

Reply:
I try adding lombok to the reproducer. The dev mode starts correctly, and if I update the class (e.g adding @Log) it works too. 
Doesn't the dev mode rely on the model to create a compile classpath that is used to recompile updated files ?

Reply:
Sure. In this method we are determining locally available projects that should be hot-reloadable. So if there is compileOnly dependency on a local project, changes to that project won't be triggering a reload. So, lombok isn't a good test in this case. You see what I mean?

Reply:
Yes I see, it looks completely logical, I will update the branch

Reply:
@aloubyansky I updated the code to take care of `compileOnly` dependency.